info/container.go

```diff
@@ -405,3 +405,36 @@ func calculateCpuUsage(prev, cur uint64) uint64 {
 	}
 	return cur - prev
 }
+
+type Percentiles struct {
+	// Indicates whether the stats are present or not.
+	Present bool `json:"present"`
+	// Average over the collected sample.
+	Mean uint64 `json:"mean"`
+	// Max seen over the collected sample.
+	Max uint64 `json:"max"`
+	// 90th percentile over the collected sample.
+	Ninety uint64 `json:"ninety"`
+}
+
+type Usage struct {
+	// Indicates amount of data available [0-100].
+	// If we have data for half a day, we'll still process DayUsage,
+	// but set PercentComplete to 50.
+	PercentComplete int32 `json:"percent_complete"`
```

Comment:
PercentComplete just indicates how many samples cadvisor has collected for that time duration. The primary use for present is when the system is configured to say not have memory isolation or cpu load, for example.

Are you concerned about cadvisor failing sometimes to collect a single resource data? We'll handle that internally in cadvisor either by ignoring the sample of extrapolating. It should not bubble up to the clients.